org.eclipse.debug.core/core/org/eclipse/debug/core/DebugPlugin.java

```diff
@@ -1280,7 +1280,7 @@ public static Document newDocument()throws CoreException {
 	 */
 	public static String serializeDocument(Document document) throws CoreException {
 		try {
-			return LaunchManager.serializeDocument(document);
+			return LaunchManager.serializeDocument(document, System.lineSeparator());
```

Comment:
The XML serializer is used in various places. In all cases, we pass the system line separator to keep current behavior. This is because in most cases these rtesources are stored in workspace metadata that is not exposed to the user. We only pass the configured line separator for the case where the data would end up in a user-specified file (specifically the launch configuration file) though this could apply to other cases in the future (where something gets stored in a user-specified location as opposed to internal workspace metadata folder).